src/App.tsx

```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  ByName = 'alphabet',
+  ByLength = 'length',
+  ByDefault = '',
+}
+
+function getVisibleGoods(
+  goods: string[],
+  sortType: SortType,
+  reverseSelected: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ByName:
+        return good1.localeCompare(good2);
+      case SortType.ByLength:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reverseSelected) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.ByDefault);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getVisibleGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+  );
+  const isResetVisible = sortType || isReversed;
+  const reset = () => {
+    setSortType(SortType.ByDefault);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortType(SortType.ByName)}
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info', {
+              'is-light': sortType !== SortType.ByName,
+            },
+          )}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortType(SortType.ByLength)}
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button',
+            'is-success', {
+              'is-light': sortType !== SortType.ByLength,
+            },
+          )}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => {
+            setIsReversed(!isReversed);
```

Comment:
```suggestion
            setIsReversed(isReversed => !isReversed);
```
When we need change state based on previous state better to use callback for this